kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -66,14 +66,15 @@ public class TransactionBuilderImpl implements TransactionBuilder {
   private final String engineInfo;
   private final Operation operation;
   private Optional<List<String>> partitionColumns = Optional.empty();
-  private Optional<List<Column>> clusteringColumns = Optional.empty();
+  private Optional<List<Column>> initialClusteringColumns = Optional.empty();
   private Optional<SetTransaction> setTxnOpt = Optional.empty();
   private Optional<Map<String, String>> tableProperties = Optional.empty();
   private Optional<Set<String>> unsetTablePropertiesKeys = Optional.empty();
   private boolean needDomainMetadataSupport = false;
 
   protected final TableImpl table;
   protected Optional<StructType> schema = Optional.empty();
+  private Optional<List<Column>> physicalClusteringColumns = Optional.empty();
```

Comment:
nit - put this next to the other private scoped variables

Reply:
also can you add a comment clarifying when this should/shouldn't be set? and it's relationship to `initialClusteringColumns`, how it's used, etc. It's a bit confusing how this is used and relates to the other variable so let's be sure to document it clearly